flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdUniqueKeys.scala

```diff
@@ -244,7 +243,25 @@ class FlinkRelMdUniqueKeys private extends MetadataHandler[BuiltInMetadata.Uniqu
       ignoreNulls: Boolean): JSet[ImmutableBitSet] = {
     val inputUniqueKeys = mq.getUniqueKeys(rel.getInput, ignoreNulls)
     val rankFunColumnIndex = RankUtil.getRankNumberColumnIndex(rel).getOrElse(-1)
-    if (rankFunColumnIndex < 0) {
+    //TODO current deduplicate on row time is still a Rank,
+    // remove this after support deduplicate on row time
+    val canConvertToDeduplicate: Boolean = {
+      val rankRange = rel.rankRange
+      val isRowNumberType = rel.rankType == RankType.ROW_NUMBER
+      val isLimit1 = rankRange match {
+        case rankRange: ConstantRankRange =>
+          rankRange.getRankStart() == 1 && rankRange.getRankEnd() == 1
+        case _ => false
+      }
+      !rel.outputRankNumber && isRowNumberType && isLimit1
```

Comment:
`!rel.outputRankNumber` is unnecessary ? because even Rank output rank number, its value is always 1